src/rvya25-profile.adoc

```diff
@@ -0,0 +1,454 @@
+
+NOTE: This profile is presented as incremental changes over RVA23.
+
+=== RVYA25U64 Profile
+
+The RVYA25U64 profile specifies the ISA features available to user-mode
+execution environments in 64-bit applications processors.  This is the
+most important profile within the application processor family in
+terms of the amount of software that targets this profile.
+
+==== RVYA25U64 Mandatory Base
+
+RVY64I is the mandatory base ISA for RVYA25U64 and is little-endian.  As
+per the unprivileged architecture specification, the `ECALL`
+instruction causes a requested trap to the execution environment.
+
+NOTE: The `ECALL` requirement is as specified in RVA23U64.
+
+==== RVYA25U64 Mandatory Extensions
+
+The following mandatory extensions are also mandatory in RVA23U64.
+
+- *M* Integer multiplication and division.
+- *A* Atomic instructions.
+- *F* Single-precision floating-point instructions.
+- *D* Double-precision floating-point instructions.
+- *C* Compressed instructions.
+- *B* Bit-manipulation instructions.
+- *Zicsr*  CSR instructions.  These are implied by presence of F.
+- *Zicntr* Base counters and timers.
+- *Zihpm* Hardware performance counters.
+- *Ziccif* Main memory regions with both the cacheability and
+  coherence PMAs must support instruction fetch, and any instruction
+  fetches of naturally aligned power-of-2 sizes up to min(ILEN,XLEN)
+  (i.e., 32 bits for RVYA25) are atomic.
+- *Ziccrse* Main memory regions with both the cacheability and coherence PMAs must support RsrvEventual.
+- *Ziccamoa* Main memory regions with both the cacheability and coherence PMAs must support all atomics in A.
+- *Zicclsm* Misaligned loads and stores to main memory regions with both the
+  cacheability and coherence PMAs must be supported.
+- *Za64rs* Reservation sets are contiguous, naturally aligned, and a
+   maximum of 64 bytes.
+- *Zihintpause* Pause hint.
+- *Zic64b* Cache blocks must be 64 bytes in size, naturally aligned in the
+address space.
+- *Zicbom* Cache-block management instructions.
+- *Zicbop* Cache-block prefetch instructions.
+- *Zicboz* Cache-Block Zero Instructions.
+- *Zfhmin* Half-precision floating-point.
+- *Zkt* Data-independent execution latency.
+- *V* Vector extension.
+- *Zvfhmin* Vector minimal half-precision floating-point.
+- *Zvbb* Vector basic bit-manipulation instructions.
+- *Zvkt* Vector data-independent execution latency.
+- *Zihintntl* Non-temporal locality hints.
+- *Zicond* Integer conditional operations.
+- *Zimop* may-be-operations.
+- *Zcmop* Compressed may-be-operations.
+- *Zcb* Additional compressed instructions.
+- *Zfa* Additional floating-Point instructions.
+- *Zawrs* Wait-on-reservation-set instructions.
+
+The following mandatory extensions are RVY specific, but may be replaced over time with binary compatible alternatives if better schemes are developed:
+
+- *Zys* Sentry definition
+- *Zylevels* Capability flow control
+
+The following mandatory extension is in addition to RVA23U64:
```

Comment:
```suggestion

The following mandatory extension is RVY specific and specifies the capability encoding format:

-* Zydefaultcap* Default capability encoding (visible to software as uycfg.base=0).

The following mandatory extensions are RVY specific, but may be replaced over time with binary compatible alternatives if better schemes are developed:

- *Zys* Sentry definition
- *Zylevels1* 2-level Capability Information Flow Control (visible to software as uycfg.features=1).

The following mandatory extension is in addition to RVA23U64:

```

Reply:
@nwf any good?

Reply:
I didn't think we were using extensions to capture encodings?  If that's fine with RVI, I think I'm happy to roll with it.